### PR TITLE
Port metrics to Prometheus

### PR DESCRIPTION
**Pull request by bgrozev**
- feat: Add a thread count metric.
- Add a stress metric.
- Move MUC stats to metrics.
- ref: Move the healthy metric to JvbHealthChecker.
- ref: Move drain and graceful shutdown metrics to VideobridgeMetrics.
- ref: Formatting.
- Extract periodic metric update into a seprate class.
- Remove conference size buckets stats
- Remove the jitter aggregate metric.
- Remove the BWE expirations stats.
- Remove "failed conferences" stats
- Remove bwe degraded/limited/controlled stats.
- Remove deprecated stats.
- tmp
- Add headers, comments.
- fix ktlint errors.
- fix: Fix histogram initialization.
- ref: Move some more metrics to VideobridgeMetrics.
- ref: Remove now unused statisticsSupplier-s.
- ref: Move more metrics to VideobridgeMetrics.
- ref: Move more metrics to VideobridgeMetrics.
- ref: Port more metrics to VideobridgeMetrics.
- ref: Remove Videobridge.Statistics.
- ref: Remove Conference.Statistics, update metrics directly.
- Add a lock around updating and accessing metrics.

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/metrics/Metrics.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.jitsi.config.JitsiConfig
+import org.jitsi.metaconfig.config
+import org.jitsi.metrics.MetricsUpdater
+import org.jitsi.videobridge.util.TaskPools
+import java.time.Duration
+
+object Metrics {
+    private val interval: Duration by config {
+        "videobridge.stats.interval".from(JitsiConfig.newConfig)
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/metrics/Metrics.kt:27]: Reading the stat in more than one place seems odd -- if you want this to be the same as `StatsManagerConfig.interval` why not just reference the object?

File: jvb/src/main/kotlin/org/jitsi/videobridge/metrics/VideobridgePeriodicMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+        if (outgoingPacketsReceived + outgoingPacketsLost > 0) {
+            outgoingLoss = outgoingPacketsLost.toDouble() / (outgoingPacketsReceived + outgoingPacketsLost)
+        }
+        var overallLoss = 0.0
+        if (incomingPacketsReceived + incomingPacketsLost + outgoingPacketsReceived + outgoingPacketsLost > 0) {
+            overallLoss = ((outgoingPacketsLost + incomingPacketsLost).toDouble()
+                    / (incomingPacketsReceived + incomingPacketsLost + outgoingPacketsReceived + outgoingPacketsLost))
+        }
+
+        // TODO LOCK
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/metrics/VideobridgePeriodicMetrics.kt:325]: Does this need resolution?

File: jvb/src/main/java/org/jitsi/videobridge/Conference.java [hunk 3]
```diff
@@ -1357,78 +1329,6 @@ public boolean isInactive()
      */
     public static class Statistics
```
Comment (JonathanLennox) [inline jvb/src/main/java/org/jitsi/videobridge/Conference.java:1330]: Does this class still remain as an empty claas after this?

File: jvb/src/main/kotlin/org/jitsi/videobridge/metrics/Metrics.kt (hunk 1 above)
Comment (bgrozev) [reply jvb/src/main/kotlin/org/jitsi/videobridge/metrics/Metrics.kt:27]: Ops, the other one was supposed to go away but I forgot it.

**Review by JonathanLennox — APPROVED** [review]

**Pull request merged**